riptide-exceptions/src/main/java/org/zalando/riptide/exceptions/TemporaryExceptionPlugin.java

```diff
@@ -0,0 +1,66 @@
+package org.zalando.riptide.exceptions;
+
+import com.google.common.base.Throwables;
+import org.springframework.http.client.ClientHttpResponse;
+import org.zalando.riptide.Plugin;
+import org.zalando.riptide.RequestArguments;
+import org.zalando.riptide.RequestExecution;
+
+import javax.net.ssl.SSLHandshakeException;
+import java.io.InterruptedIOException;
+import java.net.SocketException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.CompletionException;
+import java.util.function.Predicate;
+
+import static org.zalando.fauxpas.FauxPas.rethrow;
+import static org.zalando.fauxpas.FauxPas.throwingFunction;
+
+public final class TemporaryExceptionPlugin implements Plugin {
+
+    private final Predicate<Throwable> temporary;
+
+    public TemporaryExceptionPlugin() {
+        this(InterruptedIOException.class::isInstance,
```

Comment:
No, at least not by default. There is just too much stuff that is an IOException that is a permanent error:

- CharacterCodingException
- CharConversionException
- FileNotFoundException
- InvalidPropertiesFormatException
- MalformedURLException
- UnsupportedEncodingException

To name a few (just from the JDK alone).